dcap/types/src/quote3.rs

```diff
@@ -367,6 +367,60 @@ impl<'a> TryFrom<&'a [u8]> for CertificationData<'a> {
     }
 }
 
+impl<'a> IntoIterator for &'a CertificationData<'a> {
+    type Item = &'a [u8];
+    type IntoIter = PemIterator<'a>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        PemIterator {
+            pem_data: self.data,
+        }
+    }
+}
+
+const BEGIN_PEM: &[u8] = b"-----BEGIN ";
+const END_PEM: &[u8] = b"-----END ";
+
+struct PemIterator<'a> {
+    pem_data: &'a [u8],
+}
+
+/// Iterator over each PEM in a provided buffer.
+impl<'a> Iterator for PemIterator<'a> {
+    type Item = &'a [u8];
+    fn next(&mut self) -> Option<&'a [u8]> {
+        if self.pem_data.is_empty() {
+            return None;
+        }
+
+        let mut start = None;
+        let mut offset = 0;
+
+        // The data comes across the wire and should only use "\n" and thus
+        // won't be platform specific
+        let lines = self.pem_data.split_inclusive(|e| *e == b'\n');
```

Comment:
I looked for a PEM _walking_ crate that was `no_std` but my google fu came up empty

Reply:
Looking at `pem-rs` it seems the behavior is similar, https://github.com/jcreekmore/pem-rs/blob/master/src/parser.rs#L36
`pem-rs` will skip over another BEGIN line.  One difference is that `pem-rs` will consume mismatched END lines and then error at https://github.com/jcreekmore/pem-rs/blob/master/src/lib.rs#L163

Reply:
The ergonomics of an iterator that returns `Option<Result<T>>` rather than `Option<T>` always feel pretty clunky to me. Maybe a `parse_multi_pem()` method on `&'a CertificationData<'a>` that returns a `Result<Vec<&'a [u8]>>` which you then iterate on would be better?
If this sounds like it would produce worse code here feel free to shoot me down on this, I am not speaking with absolute confidence here.

Reply:
Definitely in a grey area here. The `type` crates we want to support `no_alloc`.  It doesn't mean we can't provide interfaces hidden behind an `alloc` feature. But _ideally_ it would be nice to find a common interface for the `alloc` and `no_alloc` use cases. Though how much time do we spend trying to make that happen versus get something working, is always subjective.

With that said anything using `Vec` would be isolated to the `alloc` feature only. We may be ok saying no alloc can use the `CertifictionData::data()` method added in #205. However raw access to `data` doesn't let one get at the PEMs in a meaningful way.  

When I initially started implementing this I tried to go straight to DER with https://docs.rs/pem-rfc7468/latest/pem_rfc7468/.  However in order to use https://docs.rs/pem-rfc7468/latest/pem_rfc7468/ in a `no_alloc` manner one must pass in the `&[u8]` buffer to contain the DER bytes. This works, but it feels clunky. Because I already had to parse the PEMs in order to feed them to https://docs.rs/pem-rfc7468/latest/pem_rfc7468/ I decided to only do the PEM iteration (at this time).
I still can't decide if the DER behavior should be available in this crate, or if it can be left to consumers.

Reply:
Ah, I didn't realize this needed to be `no_alloc` - in that case I think you're implementing this in the best way available, yeah.